docs/current/guides/757394-use-services.md

```diff
@@ -0,0 +1,347 @@
+---
+slug: /757394/use-services
+displayed_sidebar: "current"
+category: "guides"
+tags: ["go", "python", "nodejs"]
+authors: ["Alex Suraci"]
+date: "2023-03-07"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+import Embed from '@site/src/components/atoms/embed.js'
+
+# Use Services in Dagger
+
+## Introduction
+
+Dagger v0.4 introduced services, aka container-to-container networking. This feature enables users to spin up additional long-running services (as containers) and communicate with those services from their Dagger pipelines.
+
+Some common use cases for services are:
+
+- Run a test database
+- Run end-to-end integration tests
+- Run sidecar services
+
+This tutorial teaches you the basics of using services in Dagger.
+
+## Requirements
+
+This tutorial assumes that:
+
+- You have a Go, Python or Node.js development environment. If not, install [Go](https://go.dev/doc/install), [Python](https://www.python.org/downloads/) or [Node.js](https://nodejs.org/en/download/).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+
+## Key concepts
+
+Let's start with some important concepts.
+
+Dagger services are just containers, which have the following abilities:
+
+- Each container has its own network namespace and IP address
+- Each container has a unique, deterministic DNS address
+- Containers can expose ports and endpoints
+- Containers can bind other containers as services
+
+Service containers come with the following built-in features:
+
+- Service containers are started just-in-time, deduped, and stopped when no longer needed
+- Service containers are health checked prior to running clients
+- Service containers are given an alias for the client container to use as its hostname
+
+## Using hostnames
+
+Containers run in a bridge network. Each container has its own IP address that other containers can reach. Here's a simple example:
+
+<Tabs groupId="language" className="embeds">
+<TabItem value="Go">
+
+<Embed id="blm2lRNoYzE" />
```

Comment:
@vito just an FYI that because I wanted to use embeds and each embed is an independent block, I modified your code slightly by removing the common code ie. the dump() helper function and the three reusable containers, and instead writing out the entire code block in each embed. Hope this is OK. Please review and suggest further changes as needed.

Reply:
FYI easiest way to check is to use the netlify preview, which should show the code in the interactive editor

Reply:
Makes total sense :+1: